backend/cmd/worker/case_validation.go

```diff
@@ -14,16 +14,44 @@ const (
 	IdenticalCaseInBatchCode = "CASE-001"
 )
 
+var CaseRelatedTaskTypes = map[string]struct{}{
+	"family_variant_calling":     {},
+	"tumor_only_variant_calling": {},
+	//"alignment_somatic_variant_calling": {}, // NOT SUPPORTED YET
+}
+
 type CaseKey struct {
 	ProjectCode     string
 	SubmitterCaseID string
 }
 
+type StorageContext struct {
+	CasesRepo   *repository.CasesRepository
+	OrgRepo     *repository.OrganizationRepository
+	PatientRepo *repository.PatientsRepository
+	ProjectRepo *repository.ProjectRepository
+	SeqExpRepo  *repository.SequencingExperimentRepository
+	DocRepo     *repository.DocumentsRepository
+	ObsCat      *repository.ObservationCategoricalRepository
+	FamilyRepo  *repository.FamilyRepository
+	TaskRepo    *repository.TaskRepository
+}
+
 type CaseValidationRecord struct {
 	BaseValidationRecord
-	Case            types.CaseBatch
-	ProjectID       *int
-	SubmitterCaseID string
+	Context                *BatchValidationContext
+	Case                   types.CaseBatch
+	CaseID                 *int
+	ProjectID              *int
+	SubmitterCaseID        string
+	AnalysisCatalogID      *int
+	OrderingOrganizationID *int
+	DiagnosisLabID         *int
+
+	// Necessary to persist the case
+	Patients              map[string]*types.Patient
+	SequencingExperiments []*types.SequencingExperiment
+	TaskHasDocuments      []*types.TaskHasDocument
```

Comment:
TaskHasDocuments but no Document and no Task ?

Reply:
Yes, it's built dynamically in the code when handling documents in tasks. The next PR will include usage of this.